### PR TITLE
feat: aau i endorse user technical expertise - UI

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
## Description

This PR is to add UI component for user story aau i endorse user technical expertise

it contains:
- TEEndorsementModal : a model to display the endorsement options
- TEEndorsementButton: a button to trigger endorsement modal 
- AccountTEEndorsement: a controller components to render above components and provide sign logic (TODO in other PR)

Screen shots:
image

### Related ticket

Fixes #
https://github.com/MetaMask/permissionless-snaps-directory/issues/6

### Type of change

- [ ] Chore
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update

### CODE REVIEW

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+import { AccountTEEndorsement } from './AccountTEEndorsement';
+import { render } from '../../utils/test-utils';
+
+jest.mock('wagmi', () => ({
+  useEnsName: jest.fn(),
+}));
+
+const VALID_ACCOUNT_1 = '0x123';
+const VALID_ACCOUNT_2 = '0x123';
```
Comment (stanleyyconsensys) [inline src/features/account/AccountTEEndorsement.test.tsx:12]: Will update it with Leo package

**Review by satyajeetkolhapure — APPROVED** [review]

File: src/features/account/AccountTEEndorsement.test.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    await act(async () =>
+      act(() => {
+        getByText('Endorse').click();
+      }),
+    );
+
+    expect(queryByText('mock.ens.name')).toBeInTheDocument();
+  });
+
+  it('dismiss the model when sign button clicked', async () => {
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.test.tsx:90]:
```suggestion
  it('dismiss the modal when sign button clicked', async () => {
```

Comment (stanleyyconsensys) [reply src/features/account/AccountTEEndorsement.test.tsx:90]: same fault ^^

Comment (stanleyyconsensys) [reply src/features/account/AccountTEEndorsement.test.tsx]: updated

File: src/features/account/AccountTEEndorsement.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  return (
+    <>
+      <TEEndorsementButton
+        onClick={() => setShowModal(true)}
+        endorsed={false}
+      />
+      {showModal && (
+        <TEEndorsementModal
+          trustEntity={data ?? trimedAddress}
+          visability={showModal}
```
Comment (satyajeetkolhapure) [inline src/features/account/AccountTEEndorsement.tsx:53]:
```suggestion
          visibility={showModal}
```

File: src/features/account/components/modals/TEEndorsementModal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    const onSign = jest.fn().mockResolvedValue(true);
+    const onClose = jest.fn();
+
+    const { queryByText } = render(
+      <TEEndorsementModal
+        trustEntity="mock.ens.name"
+        options={options}
+        onClose={onClose}
+        onSign={onSign}
+        visability={true}
```
Comment (satyajeetkolhapure) [inline src/features/account/components/modals/TEEndorsementModal.test.tsx:30]:
```suggestion
        visibility={true}
```

File: src/features/account/components/modals/TEEndorsementModal.test.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      value: option.value,
+    }));
+
+    const { queryByText } = render(
+      <TEEndorsementModal
+        trustEntity="mock.ens.name"
+        options={optionsWithoutDescription}
+        onClose={onClose}
+        onSign={onSign}
+        visability={true}
```
Comment (satyajeetkolhapure) [inline src/features/account/components/modals/TEEndorsementModal.test.tsx:56]:
```suggestion
        visibility={true}
```

File: src/features/account/components/modals/TEEndorsementModal.test.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    const onSign = jest.fn().mockResolvedValue(true);
+    const onClose = jest.fn();
+
+    const { getByText } = render(
+      <TEEndorsementModal
+        trustEntity="mock.ens.name"
+        options={options}
+        onClose={onClose}
+        onSign={onSign}
+        visability={true}
```
Comment (satyajeetkolhapure) [inline src/features/account/components/modals/TEEndorsementModal.test.tsx:77]:
```suggestion
        visibility={true}
```

File: src/features/account/components/modals/TEEndorsementModal.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+import { useState, type FunctionComponent } from 'react';
+
+import { AvatarReportBlueIcon, RequestSignModal } from '../../../../components';
+
+export type TEEndorsementModalProps = {
+  trustEntity: string;
+  options: TEEndorsementOption[];
+  onSign: (arg: string[]) => Promise<void>;
+  onClose: () => void;
+  visability: boolean;
```
Comment (satyajeetkolhapure) [inline src/features/account/components/modals/TEEndorsementModal.tsx:12]:
```suggestion
  visibility: boolean;
```

File: src/features/account/components/modals/TEEndorsementModal.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  value: string;
+  description?: string;
+};
+
+export const TEEndorsementModal: FunctionComponent<TEEndorsementModalProps> = ({
+  trustEntity,
+  options,
+  onSign,
+  onClose,
+  visability,
```
Comment (satyajeetkolhapure) [inline src/features/account/components/modals/TEEndorsementModal.tsx:26]:
```suggestion
  visibility,
```

File: src/features/account/components/modals/TEEndorsementModal.tsx [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+    });
+
+    onSign(args).finally(() => {
+      setIsLoading(false);
+    });
+  };
+
+  return (
+    <RequestSignModal
+      isOpen={visability}
```
Comment (satyajeetkolhapure) [inline src/features/account/components/modals/TEEndorsementModal.tsx:53]:
```suggestion
      isOpen={visibility}
```

**Review by ritave — APPROVED** [review]
Fix `visibility` typo and good to go

**Review by dawnseeker8 — APPROVED** [review]

**Pull request merged**